src/main/java/com/aws/greengrass/localdebugconsole/SimpleHttpServer.java

```diff
@@ -100,35 +127,72 @@ public SimpleHttpServer(Topics t, Kernel kernel, DeviceConfiguration deviceConfi
     public void postInject() {
         super.postInject();
         config.lookup(CONFIGURATION_CONFIG_KEY, "port").dflt(port).subscribe((w, n) -> {
+            int oldPort = port;
             port = Coerce.toInt(n);
             if (port < 1000) {
                 logger.atWarn().kv("port", port).kv("defaultPort", DEFAULT_HTTP_PORT)
                         .log("Port number should not be smaller than 1000. Using default.");
                 port = DEFAULT_HTTP_PORT;
             }
-            // TODO: should restart server on new port
+            if (oldPort != port) {
+                requestRestart();
+            }
+        });
+        config.lookup(CONFIGURATION_CONFIG_KEY, "httpsEnabled").dflt(DEFAULT_HTTPS_ENABLED).subscribe((w, n) -> {
+            boolean oldEnabled = httpsEnabled;
+            httpsEnabled = Coerce.toBoolean(n);
+            if (oldEnabled != httpsEnabled) {
+                requestRestart();
+            }
         });
         config.lookup(CONFIGURATION_CONFIG_KEY, "websocketPort").dflt(websocketPort).subscribe((w, n) -> {
+            int oldPort = websocketPort;
             websocketPort = Coerce.toInt(n);
             if (port < 1000) {
```

Comment:
this is the first time I'm looking at this - but shouldn't this be checking if `websocketPort < 1000` not `port < 1000` since that is checked above and the ports are used differently?

Reply:
Good call, will fix.